### ISSUE TITLE
More sensible default optimizer settings

### ISSUE THREAD

**Issue by odelalleau** [NVIDIA/NeMo-Aligner#12]
Current default settings have no proper justification and makes one wonder how they were chosen and to which extent they matter. Draft suggestion (this is a template based on RM trainer, for instance the PPO learning rate may be 10x lower):

```yaml
  optim:
    name: distributed_fused_adam
    bucket_cap_mb: 200
    overlap_grad_sync: False
    contiguous_grad_buffer: True
    lr: 1e-5  # instead of 9e-6 (looks better)
    weight_decay: 0.01  # instead of 0.1 -- experiments show it doesn't seem to really matter anyway, so better keep it small
    betas: 
    - 0.9
    - 0.999  # instead of 0.98 -- PyTorch default -- needs to be tested to be sure it has no major impact
    sched:
      name: CosineAnnealing
      warmup_steps: 0  # instead of 10 -- to be tested if it matters for SFT / RM / DPO / critic -- I expect only the PPO actor should potentially benefit from warmup while the critic is training
      constant_steps: 0  # instead of 1000 which can make the lr flatline at `min_lr` for some settings
      min_lr: 1e-6  # instead of 9e-7 (looks better)
```

This will definitely require some testing but I think it will be less confusing that way.
